t/plugin/azure-functions.t

```diff
@@ -0,0 +1,372 @@
+--- config
+    location /t {
+        content_by_lua_block {
+            local t = require("lib.test_admin").test
+
+            local code, body = t('/apisix/admin/routes/1',
+                 ngx.HTTP_PUT,
+                 [[{
+                        "plugins": {
+                            "azure-functions": {
+                                "function_uri": "http://localhost:8765/httptrigger"
+                            }
+                        },
+                        "upstream": {
+                            "nodes": {
+                                "127.0.0.1:1982": 1
+                            },
+                            "type": "roundrobin"
+                        },
+                        "uri": "/azure"
+                }]],
+                [[{
+                    "node": {
+                        "value": {
+                            "plugins": {
+                                "azure-functions": {
+                                    "keepalive": true,
+                                    "timeout": 3000,
+                                    "ssl_verify": true,
+                                    "keepalive_timeout": 60000,
+                                    "keepalive_pool": 5,
+                                    "function_uri": "http://localhost:8765/httptrigger"
+                                }
+                            },
+                            "upstream": {
+                                "nodes": {
+                                    "127.0.0.1:1982": 1
+                                },
+                                "type": "roundrobin"
+                            },
+                            "uri": "/azure"
+                        },
+                        "key": "/apisix/routes/1"
+                    },
+                    "action": "set"
+                }]]
+                )
+
+            if code >= 300 then
+                ngx.status = code
+                ngx.say("fail")
+                return
+            end
+
+            ngx.say(body)
+        }
+    }
+--- response_body
+passed
+
+
+
+=== TEST 4: Test plugin endpoint
+--- config
+    location /t {
+        content_by_lua_block {
+            local t = require("lib.test_admin").test
+            local core = require("apisix.core")
+
+            local code, _, body, headers = t("/azure", "GET")
+             if code >= 300 then
+                ngx.status = code
+                ngx.say(body)
+                return
+            end
+
+            -- headers proxied 2 times -- one by plugin, another by this test case
+            core.response.set_header(headers)
+            ngx.print(body)
+        }
+    }
+--- response_body
+faas invoked
+--- response_headers
+Content-Length: 13
+X-Extra-Header: MUST
+
+
+
+=== TEST 5: http2 check response body and headers
+--- http2
+--- exec
+curl -XGET --http2 --http2-prior-knowledge localhost:1984/azure
```

Comment:
We can use regular `--- request` to send http2 request, see
https://github.com/openresty/test-nginx/blob/9d5a20f884d67974f9eafeed5a1540cd32945fb2/lib/Test/Nginx/Socket.pm#L2310

Reply:
Nice, it is. Updated